packages/shared/lib/core/ledger/utils/isLedgerDeviceMatchingActiveProfile.ts

```diff
@@ -0,0 +1,30 @@
+import { get } from 'svelte/store'
+
+import { selectedAccount } from '@core/account/stores'
+import { profileManager } from '@core/profile-manager/stores'
+
+import { LedgerAppName } from '../enums'
+import { isLedgerAppOpen } from './isLedgerAppOpen'
+
+export async function isLedgerDeviceMatchingActiveProfile(): Promise<boolean | undefined> {
+    if (isLedgerAppOpen(LedgerAppName.Shimmer)) {
+        try {
+            const account = get(selectedAccount)
```

Comment:
Use `getSelectedAccount()` instead of `get(selectedAccount)` to decouple from svelte stores. Ideally, this should be done for the other functions as well.